### ISSUE TITLE
restrict CIDR range for openvpn-server and bastion host

### ISSUE THREAD

**Issue by janwillies** [gardener/gardener#1218]
**What would you like to be added**:
I would like to limit the CIDR ranges which are allowed to connect to the openvpn-server in the shoot, and the bastion host.

**Why is this needed**:
Currently the openvpn-server in the shoot cluster is open to `0.0.0.0/0`, as is the bastion host. While CIDR limits are often implemented as a security consideration, they have other benefits as well: blocking unwanted requests early thus keeping logging and monitoring databases clean, lower the load on backend service, etc

**Comment by vasu1124** [comment]
We are working/waiting on an upsteam KEP:
https://github.com/kubernetes/enhancements/issues/1217
https://github.com/kubernetes/enhancements/pull/1216
https://github.com/kubernetes/kubernetes/pull/79312

While these will help reduce the number of control plane LB IPs, depending on the provider these IPs cannot be considered stable. But reducing the CIDR will also limit effective DR (we can recover a control plane automatically on a different seed!). We recommend zero trust/beyond corp setups.

Gardenlet https://github.com/gardener/gardener/issues/1576 will offer new options to restrict the CIDR.

**Comment by rfranzke** [comment]
There is a GEP for inverting the VPN direction, https://github.com/gardener/gardener/pull/2170, though, it will only be implemented for new Kubernetes versions. Older versions will still use the VPN solution.

**Comment by rfranzke** [comment]
@DockToFuture @ScheererJ will this still be necessary with your new reversed VPN proposal #3812?

**Comment by ScheererJ** [comment]
It depends on how you look at the problem. With #3771/#3812, the openvpn server moves from the shoot cluster to the shoot control plane in the seed cluster. Therefore, if this is "only" about having clean logs/restricted access vector to the shoot cluster then it is no longer relevant. However, even with #3771/#3812 we continue to have an openvpn server, even though it is now located in the shoot control plane in the seed cluster. We still do not restrict connections to the openvpn server to certain cidr ranges. Hence, it might still be relevant.

**Comment by rfranzke** [comment]
Thanks @ScheererJ, do you or @DockToFuture plan to take this in the near future?

**Comment by ScheererJ** [comment]
We do not plan to implement this for the reversed VPN proposal. Meanwhile we have added an authorization server for the reversed vpn solution. It would be fairly straight forward to implement the filter there (https://github.com/gardener/ext-authz-server/blob/master/pkg/auth/v3/auth.go).
@janwillies Feel free to propose a corresponding change if you still need this.
